core/build.gradle.kts

```
@@ -187,6 +194,65 @@ ide {
     generatedSource(javaCCTest, "test")
 }
 
+fun configureAnnotationSet(task: JavaCompile, sourceSet: SourceSet) {
+    with(task) {
+        source = sourceSet.java
+        classpath = sourceSet.compileClasspath
+        destinationDirectory.set(options.generatedSourceOutputDirectory)
+        options.compilerArgs.add("-proc:only")
+        org.gradle.api.plugins.internal.JvmPluginsHelper.configureAnnotationProcessorPath(sourceSet, sourceSet.java, options, project)
+    }
+}
+
+val annotationProcessorMain by tasks.registering(JavaCompile::class) {
+    dependsOn(javaCCMain)
+    configureAnnotationSet(this, sourceSets.main.get())
```

If you use an extension function, then `this` would be passed automatically.

```suggestion
    configureAnnotationSet(sourceSets.main.get())
```